### PR TITLE
Add core profiler user profile page

### PR DESCRIPTION
**Pull request by chihsuan**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/team-ghidorah/issues/196.

This PR adds Core Profiler "Which one of these best describes you?" page.

figma: Y5pUYSJPsGEud1vknUZhi8-fi-739_52372

01  User's profile - Beginner

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Scenario: User enters the user profile page**

1. Install and activate WooCommerce on a brand new site
2. Install the `WooCommerce Beta Tester` plugin
3. Go to /wp-admin/tools.php?page=woocommerce-admin-test-helper
4. Enable the 'core-profiler' feature flag
5. Open your browser developer console and run `localStorage.setItem( 'debug', 'wc-admin:*' );` to set debug for tracks
6. Visit http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fsetup-wizard
7. Clicks on the `Set up my store` button
8. Observe that `Which one of these best describes you?` page is shown
9. Confirm the UI matches the design. (Y5pUYSJPsGEud1vknUZhi8-fi-739_52372)
10. Open your browser developer console and confirm `storeprofiler_step_view` track is recorded with  prop step = user_profile.

**Scenario: User chooses Yes, I'm selling online or I'm selling both online and offline**

1. On Which one of these best describes you?` page, choose `I'm already selling`
2. Chooses either `Yes, I'm selling online` or `I'm selling both online and offline`
3. Then render [Which platform(s) are you currently using?](Y5pUYSJPsGEud1vknUZhi8-fi-739_59745) dropdown
4. Test the dropdowns

**Scenario: User submits the form**

1. On Which one of these best describes you?` page, clicks on `Continue` button
2. Open your browser developer console and confirm `storeprofiler_step_complete` track is recorded with prop step = user_profile.
3. Open your browser developer console and confirm that the `wcadmin_storeprofiler_user_profile` track is recorded in your chosen options.
4. Run `wp option get woocommerce_onboarding_profile`
5. Confirm the choices should be saved to woocommerce_onboarding_profile option

**Scenario: Restore previous user selection**

1. Visit http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fsetup-wizard again
2. Clicks on the `Set up my store` button
3. The radio button choice should be restored as the previous selection.

**Scenario: Skip the step**

1. Clicks on the `Skip this step` button
2. Confirmed storeprofiler_store_details_skip track is recorded
3. Run `wp option get woocommerce_onboarding_profile`
4. Should returns 

```
array (
  'skipped' => true,
)
```
Screenshot 2023-05-17 at 12 13 50

### CODE REVIEW

File: plugins/woocommerce-admin/client/core-profiler/pages/UserProfile.tsx [hunk 1]
```diff
@@ -1,41 +1,282 @@
...
+		useState< Array< SellingPlatform > | null >(
+			context.userProfile.sellingPlatforms || null
+		);
+
+	const renderAlreadySellingOptions = () => {
+		return (
+			<>
+				<div className="woocommerce-profiler-selling-online-question">
+					<p className="woocommerce-profiler-question-label">
+						{ __( 'Are your selling online?', 'woocommerce' ) }
```
Comment (rjchow) [inline plugins/woocommerce-admin/client/core-profiler/pages/UserProfile.tsx]:
```suggestion
						{ __( 'Are you selling online?', 'woocommerce' ) }
```
Think it's a typo in the figma

**Review by moon0326 — COMMENTED** [review]
Looking good 🎉 

I have two minor CSS change requests.

1. H1 isn't aligned correctly according to Figma

**Figma**:
Screen Shot 2023-05-17 at 7 17 35 PM

**This PR**:
Screen Shot 2023-05-17 at 7 15 08 PM

2. Dropdown hover color and empty space.
 
Screen Shot 2023-05-17 at 7 15 47 PM

Everything else LGTM.

**Comment by chihsuan** [comment]
> I have two minor CSS change requests.
> 
> 1. H1 isn't aligned correctly according to Figma
> 2. Dropdown hover color and empty space.

Nice catch! @moon0326  Fixed in 9f40dacbe279bb3e6673a35bba02262c1a57333d and e6284bf7af1e8c6afa9d9368ce7d147413567785

Screenshot 2023-05-18 at 11 20 53


Also addressed the input box issue that was identified in the design call.


Screenshot 2023-05-18 at 11 07 44

**Review by moon0326 — DISMISSED** [review]
Thank you for addressing the changes so quickly! 

LGTM and tested well 🚀

**Comment by rjchow** [comment]
@chihsuan , @moon0326 
I noticed that we're using different options compared to the current OBW, not sure if this is intended so just asking -
https://github.com/woocommerce/woocommerce/blob/6227ecb6f2c63caf5340f292c2d64086840d8b95/plugins/woocommerce-admin/client/profile-wizard/steps/store-details/index.js#L488

If we look at the shape of the object for woocommerce_onboarding_profile:
```
{
  "completed": null,
  "skipped": null,
  "industry": null,
  "product_types": null,
  "product_count": null,
  "selling_venues": null,
  "number_employees": null,
  "revenue": null,
  "other_platform": null,
  "other_platform_name": null,
  "business_extensions": null,
  "theme": null,
  "setup_client": null,
  "wccom_connected": false,
  "is_agree_marketing": null,
  "store_email": null,
  "is_store_country_set": null
}
```

And the API schema here: https://github.com/woocommerce/woocommerce/blob/6227ecb6f2c63caf5340f292c2d64086840d8b95/plugins/woocommerce/src/Admin/API/OnboardingProfile.php#L262

I'm not really sure if these values are used elsewhere

**Comment by moon0326** [comment]
> @chihsuan , @moon0326 I noticed that we're using different options compared to the current OBW, not sure if this is intended so just asking - https://github.com/woocommerce/woocommerce/blob/6227ecb6f2c63caf5340f292c2d64086840d8b95/plugins/woocommerce-admin/client/profile-wizard/steps/store-details/index.js#L488
> 
> If we look at the shape of the object for woocommerce_onboarding_profile:
> 
> ```
> {
>   "completed": null,
>   "skipped": null,
>   "industry": null,
>   "product_types": null,
>   "product_count": null,
>   "selling_venues": null,
>   "number_employees": null,
>   "revenue": null,
>   "other_platform": null,
>   "other_platform_name": null,
>   "business_extensions": null,
>   "theme": null,
>   "setup_client": null,
>   "wccom_connected": false,
>   "is_agree_marketing": null,
>   "store_email": null,
>   "is_store_country_set": null
> }
> ```
> 
> And the API schema here: https://github.com/woocommerce/woocommerce/blob/6227ecb6f2c63caf5340f292c2d64086840d8b95/plugins/woocommerce/src/Admin/API/OnboardingProfile.php#L262
> 
> I'm not really sure if these values are used elsewhere

Good catch!

I think I missed them when I reviewed the onboarding options. What do you think about merging it now and following up in a separate issue? We also should go over the options on other pages.

**Comment by chihsuan** [comment]
> Good catch!

+1 @rjchow 👍 

> I think I missed them when I reviewed the onboarding options. What do you think about merging it now and following up in a separate issue? We also should go over the options on other pages.

I agree. It seems that some options are no longer needed, and we also need to modify the profile items API.

**Review by ilyasfoo — COMMENTED** [review]
Awesome work, @chihsuan!

It seems the dropdown popup would overlap the dropdown element above it when users screen height is limited. However, I think this is a low priority bug. We could handle it in a follow-up if you think it's necessary.

https://github.com/woocommerce/woocommerce/assets/3747241/db34c344-b9d7-4dcf-91ef-805459df2a58

Also, on screens with around 678px width, there are a few UI glitches:

1. Extra space before loading (I think this is not related to this PR, we can ignore it for this PR)
2. Form elements aren't aligned (perhaps we should center them)

image

**Comment by rjchow** [comment]
> What do you think about merging it now and following up in a separate issue? We also should go over the options on other pages.

Yep ok, let's do that!

**Comment by chihsuan** [comment]
> It seems the dropdown popup would overlap the dropdown element above it when users screen height is limited. However, I think this is a low priority bug. We could handle it in a follow-up if you think it's necessary.

> Also, on screens with around 678px width, there are a few UI glitches:

> Extra space before loading (I think this is not related to this PR, we can ignore it for this PR)
> Form elements aren't aligned (perhaps we should center them)

@ilyasfoo Good catch! Fixed alignment issue in aa429b83d6278fe1a44758eb9c80310413883188 and dropdown issue in 7f881ac53c7a7775bb5ce21836912afb2cf09b75.

However,  it seems that the scrollIntoView function doesn't work very well. But I think I'll leave it as a follow-up.

https://github.com/woocommerce/woocommerce/assets/4344253/d0b8e8d4-b0ed-4824-89fd-cac4d5394958

**Review by ilyasfoo — APPROVED** [review]
Thanks for addressing those issues, it looks good now! 🚢

**Pull request merged**